cwl/src/main/scala/cwl/CwlInputCoercion.scala

```diff
@@ -0,0 +1,84 @@
+package cwl
+
+import cats.instances.list._
+import cats.syntax.traverse._
+import cats.syntax.validated._
+import cats.syntax.option._
+import common.validation.ErrorOr.ErrorOr
+import common.validation.Validation._
+import cwl.File.FileOrDirectory
+import io.circe.{Json, JsonNumber, JsonObject}
+import shapeless.Coproduct
+import wom.executable.Executable.DelayedCoercionFunction
+import wom.types._
+import wom.values._
+
+// With recursive types we could let circe parse it for us, but until we figure that out just parse it as Json and
+// manually check for File / Directory structures
+private [cwl] object CwlInputCoercion extends Json.Folder[DelayedCoercionFunction] {
+
+  // TODO: This should handle directories
+  def toFile(json: Json): Option[File] = {
+    json.asObject.map( obj => {
+      val values = obj.toMap
+      def getStringValue(key: String) = values.get(key).flatMap(_.asString)
+
+      val secondaryFiles: Option[Array[FileOrDirectory]] = values.get("secondaryFiles").flatMap({ json =>
+        for {
+          array <- json.asArray
+          asFiles = array.flatMap(toFile).toArray.map(Coproduct[FileOrDirectory](_))
```

Comment:
NOTE: Depending on who merges first, this should be a call to an intermediate `toFileOrDirectory` that checks to see if the json object is a `File` or `Directory` just like `onObject` below.